src/workerd/jsg/value.h

```diff
@@ -445,31 +445,34 @@ class NameWrapper {
 // same reason discussed in PrimitiveWrapper.
 class StringWrapper {
  public:
-  static constexpr const char* getName(kj::String*) {
-    return "string";
-  }
-
   // TODO(someday): The conversion to kj::String doesn't explicitly consider the distinction
   // between DOMString (~ WTF-8; could contain invalid code points) and USVString (invalid code
   // points are always replaced with U+FFFD). Code should make an explict choice between the two.
 
-  static constexpr const char* getName(kj::ArrayPtr<const char>*) {
-    return "string";
-  }
-  static constexpr const char* getName(kj::Array<const char>*) {
+  template <typename T>
+    requires(kj::isSameType<T, kj::ArrayPtr<const char>>() ||
+        kj::isSameType<T, kj::Array<const char>>() || kj::isSameType<T, kj::String>())
```

Comment:
All this `kj::isSameType` business has me thinking we need to invent `kj::oneOfType<T,  kj::ArrayPtr<const char>>, kj::Array<const char>, kj::String>>()`